### DISCUSSION TITLE
Using  url = "http://localip:7851/api/tts-generate, sound is only playing on the server

### DISCUSSION THREAD

**Discussion by POD-io** [erew123/alltalk_tts#250] [Q&A] [answered]
Sending a call to the server running alltalk_tts, and wrote a program to call it from a client machine on my network. However, when I do so, the audio is playing on my server instead of the client that is calling it.

Any ideas what is causing this?

**Comment by erew123** [comment, marked as answer]
Hi @POD319 

Please make sure the call has autoplay=false

🟠 autoplay: Enable or disable playing the generated TTS to your standard sound output device at time of TTS generation.

-d "autoplay=true"
-d "autoplay=false"

**Reply by POD-io** [reply]
> Hi @POD319
> 
> Please make sure the call has autoplay=false
> 
> 🟠 autoplay: Enable or disable playing the generated TTS to your standard sound output device at time of TTS generation.
> 
> -d "autoplay=true" -d "autoplay=false"

This was an interesting way to find out I can 'remotely' play sounds on my server. However my use-case is intended to play the audio locally on the client that is making the call. Is there anything I can do for that, or is that simply playing the file that is generated?

**Reply by erew123** [reply]
The answer is yes, but it depends on what your program is built/based on (python, javascript, batch file etc). When you generate the TTS AllTalk is responding with the paths to the audio https://github.com/erew123/alltalk_tts?tab=readme-ov-file#-tts-generation-response

```
{"status":"generate-success","output_file_path":"C:\\text-generation-webui\\extensions\\alltalk_tts\\outputs\\myoutputfile_1704141936.wav","output_file_url":"http://127.0.0.1:7851/audio/myoutputfile_1704141936.wav","output_cache_url":"http://127.0.0.1:7851/audiocache/myoutputfile_1704141936.wav"}
```

You just need something within your code that will play it from the URL. So it literally depends on the coding language you are using. For example if you are on Python you may use sounddevice https://python-sounddevice.readthedocs.io/en/0.4.7/. HTML you would use an audioplayer (possibly with autoplay set in the code). etc..
